src/commands/createFunction/createFunction.ts

```diff
@@ -54,3 +43,42 @@ export async function createFunction(
     await wizard.prompt(actionContext);
     await wizard.execute(actionContext);
 }
+
+async function getWorkspacePath(actionContext: IActionContext): Promise<string> {
+    let folder: WorkspaceFolder | undefined;
+    if (!workspace.workspaceFolders || workspace.workspaceFolders.length === 0) {
+        const message: string = localize('noWorkspaceWarning', 'You must have a project open to create a function.');
+        const newProject: MessageItem = { title: localize('createNewProject', 'Create new project') };
+        const openExistingProject: MessageItem = { title: localize('openExistingProject', 'Open existing project') };
+        const result: MessageItem = await ext.ui.showWarningMessage(message, { modal: true }, newProject, openExistingProject);
+
+        if (result === newProject) {
+            // don't wait
+            commands.executeCommand('azureFunctions.createNewProject');
+            actionContext.properties.noWorkspaceResult = 'createNewProject';
+        } else {
+            const uri: Uri[] = await ext.ui.showOpenDialog({
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                openLabel: localize('open', 'Open')
+            });
+            // don't wait
+            commands.executeCommand('vscode.openFolder', uri[0]);
+            actionContext.properties.noWorkspaceResult = 'openExistingProject';
+        }
+
+        actionContext.suppressErrorDisplay = true;
+        throw new NoWorkspaceError();
```

Comment:
If you've opened  project would you still have to throw NoWorkspaceError?

Reply:
Opening a folder restarts the extension host, so they have to re-start creating a function. I'm just throwing the error so I can track it in telemetry